### ISSUE TITLE
inline-c-cpp fails to build on OS X (missing stdc++)

### ISSUE THREAD

**Issue by basvandijk** [NixOS/nixpkgs#16213]
## Issue description
`haskellPackages.inline-c-cpp` fails to build on OS X.

### Steps to reproduce
```
nix-build -A haskellPackages.inline-c-cpp ~/.nix-defexpr/channels/nixpkgs
...
Setup: Missing dependency on a foreign library:
* Missing C library: stdc++
```

This seems similar to: https://github.com/NixOS/nix/issues/872.

## Technical details

* System: nixpkgs-unstable on x86_64-darwin.
* Nix version: nix-env (Nix) 1.11.2
* Nixpkgs version: 16.09pre83147.df89584

Pinging @peti.

**Comment by peti** [comment]
`cabal2nix` drops dependencies on `stdc++` because I'm not sure which package should be linked to fulfill them. Normally, the compiler provides that library, but apparently that's different on Darwin?

**Comment by matthewbauer** [comment]
Darwin has it but it's called "libc++" instead of "libstdc++". They should be approximately compatible I think. Elsewhere we just replace libstdc++ with libc++ on Darwin. like in https://github.com/NixOS/nixpkgs/blob/22ae90ae90c1cf9da2d838e6194b253a27f916ef/pkgs/development/haskell-modules/configuration-common.nix#L236

I wonder if there is a better way?

**Comment by basvandijk** [comment]
This will be fixed once #16262 is merged.
